tests/gui/test_logic.py

```diff
@@ -0,0 +1,100 @@
+import platform
+import subprocess
+from unittest import mock
+
+import pytest
+
+from dangerzone.gui.logic import DangerzoneGui
+
+if platform.system() == "Linux":
+    from xdg.DesktopEntry import DesktopEntry
+
+
+@pytest.mark.skipif(platform.system() != "Linux", reason="Linux-only test")
+def test_order_mime_handers() -> None:
+    """
+    Given a default mime handler returned by xdg-mime,
+    ensure it is the first item available in the list
+    of compatible applications.
+    """
+    mock_app = mock.MagicMock()
+    dummy = mock.MagicMock()
+
+    mock_desktop = mock.MagicMock(spec=DesktopEntry)
+    mock_desktop.getMimeTypes.return_value = "application/pdf"
+    mock_desktop.getExec.side_effect = [
```

Comment:
Great q :)  if you use `return_value` you get the same value each time you access the mock. If you use `side_effect` and pass a list like this, you cycle through the values in the list one at a time each time the mock method is called.  

This test simulates the `os.listdir` calls, which look in [a few different places](https://github.com/freedomofpress/dangerzone/blob/65776d8c0506535c20f74f4375193287e1e8a78b/dangerzone/gui/logic.py#L101-L104) for available .desktop files, and each time, returns a different filename. (If using `return_value` eg with a single dummy value, then [here](https://github.com/freedomofpress/dangerzone/blob/65776d8c0506535c20f74f4375193287e1e8a78b/dangerzone/gui/logic.py#L116-L117) the entries would just be overwritten rather than added to the dict since the name is the key.)   

I could have used random values like random uuid, random string, whatever, instead of side_effect, but it seemed more readable this way.